src/Symfony/Component/Dotenv/Command/DebugCommand.php

```diff
@@ -125,19 +128,36 @@ private function getEnvFiles(): array
         return $files;
     }
 
-    private function getFilePath(string $file): string
+    private function getFilePath(?string $dotenvDirectory, string $file): string
     {
-        return $this->projectDirectory.\DIRECTORY_SEPARATOR.$file;
+        return $this->projectDirectory.(null === $dotenvDirectory ? '' : \DIRECTORY_SEPARATOR.$dotenvDirectory).\DIRECTORY_SEPARATOR.$file;
     }
 
-    private function loadValues(string $file): array
+    private function loadValues(?string $dotenvDirectory, string $file): array
     {
-        $filePath = $this->getFilePath($file);
+        $filePath = $this->getFilePath($dotenvDirectory, $file);
 
         if (str_ends_with($filePath, '.php')) {
             return include $filePath;
         }
 
         return (new Dotenv())->parse(file_get_contents($filePath));
     }
+
+    private function getDotenvDirectory(): ?string
```

Comment:
The changes can be simplified if this method returns the full value of `dotenv_path`, defaulting to `.env`.

The method signature becomes:
```php
private function getDotenvFile(): string
```

The basename cannot be ignored if someone something different than `.env` as filename. 

```json
{
    "...": "...",
    "extra": {
        "...": "...",
        "runtime": {
            "dotenv_path": "my/custom/path/to/custom.env"
        }
    }
}
```